src/detect-noalert.c

```diff
@@ -20,33 +20,43 @@
  *
  * \author Victor Julien <victor@inliniac.net>
  *
- * Implements the noalert keyword
+ * Implements the noalert and alert keywords.
  */
 
 #include "suricata-common.h"
+#include "action-globals.h"
 #include "detect.h"
 #include "detect-noalert.h"
 #include "util-debug.h"
+#include "util-validate.h"
 
-static int DetectNoalertSetup (DetectEngineCtx *, Signature *, const char *);
+static int DetectNoalertSetup(DetectEngineCtx *de_ctx, Signature *s, const char *nullstr)
+{
+    DEBUG_VALIDATE_BUG_ON(nullstr != NULL);
+
+    s->action &= ~ACTION_ALERT;
+    return 0;
+}
+
+static int DetectAlertSetup(DetectEngineCtx *de_ctx, Signature *s, const char *nullstr)
+{
+    DEBUG_VALIDATE_BUG_ON(nullstr != NULL);
+
+    s->action |= ACTION_ALERT;
+    return 0;
+}
 
-void DetectNoalertRegister (void)
+void DetectNoalertRegister(void)
 {
     sigmatch_table[DETECT_NOALERT].name = "noalert";
     sigmatch_table[DETECT_NOALERT].desc = "no alert will be generated by the rule";
     sigmatch_table[DETECT_NOALERT].url = "/rules/flow-keywords.html";
     sigmatch_table[DETECT_NOALERT].Setup = DetectNoalertSetup;
     sigmatch_table[DETECT_NOALERT].flags |= SIGMATCH_NOOPT;
-}
 
-static int DetectNoalertSetup (DetectEngineCtx *de_ctx, Signature *s, const char *nullstr)
-{
-    if (nullstr != NULL) {
-        SCLogError("nocase has no value");
-        return -1;
-    }
-
-    s->flags |= SIG_FLAG_NOALERT;
-    return 0;
+    sigmatch_table[DETECT_ALERT].name = "alert";
+    sigmatch_table[DETECT_ALERT].desc = "alert will be generated by the rule";
+    sigmatch_table[DETECT_ALERT].url = "/rules/flow-keywords.html";
```

Comment:
Looks like we need to document this new behavior, and possibly also properly document `noalert`. We do explain it, but only in passing, when talking about other keywords.

Reply:
Added ticket to track this: https://redmine.openinfosecfoundation.org/issues/6685